shell/platform/linux/public/flutter_linux/fl_texture.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_TEXTURE_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_TEXTURE_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include <glib-object.h>
+#include <stdint.h>
+
+G_BEGIN_DECLS
+
+G_DECLARE_DERIVABLE_TYPE(FlTexture, fl_texture, FL, TEXTURE, GObject)
+
+/**
+ * FlTexture:
+ *
+ * #FlTexture is an abstract class that represents an OpenGL texture.
+ *
+ * If you want to render textures in other OpenGL context, create and use the
+ * #GdkGLContext by calling gdk_window_create_gl_context () with the #GdkWindow
+ * of #FlView. The context will be shared with the one used by Flutter.
+ */
+
+struct _FlTextureClass {
+  GObjectClass parent_class;
+
+  /**
+   * Virtual method called when Flutter populates this texture. The OpenGL
+   * context used by Flutter has been already set.
+   * @texture: an #FlTexture.
+   * @target: texture target (example GL_TEXTURE_2D or GL_TEXTURE_RECTANGLE).
```

Comment:
Why is this necessary? We should not be binding the entire API structure to a specific rendering backend.

Reply:
According to the existing variable name defined in `embedder.h`, It's totally texture id.
https://github.com/flutter/engine/blob/f1d78fd48aeff9c61274a2abf804583e14cb3ce9/shell/platform/embedder/embedder.h#L267

Reply:
Actually we may come up with a better API for compatibility with Vulkan.

Reply:
You're linking to an internal implementation class. We can put whatever we want there, and change it at will without breaking anyone.

This comment thread is on **public API** that we will be suck with. The public Windows API has an agnostic base class: https://github.com/flutter/engine/blob/f1d78fd48aeff9c61274a2abf804583e14cb3ce9/shell/platform/common/client_wrapper/include/flutter/texture_registrar.h

Reply:
These APIs only accept pixel buffers, instead of allowing plugin to register a hardware-accelerated texture. We are taking raw OpenGL/Vulkan textures into consideration here. So exposing OpenGL related information is necessary. But we can make another subclass `FlTextureGL` to allow extension.

Reply:
> These APIs only accept pixel buffers, instead of allowing plugin to register a hardware-accelerated texture.

The pixel-buffer-specific texture class only accepts pixel buffers. The overall API is generic, because it it's based on this type:
```
// The available texture variants.
// Only PixelBufferTexture is currently implemented.
// Other variants are expected to be added in the future.
typedef std::variant<PixelBufferTexture> TextureVariant;
```

> We are taking raw OpenGL/Vulkan textures into consideration here.

There is a difference between "taking into consideration" and "requiring for all current and future texture implementations, with no allowance for the possibility of a non-OpenGL-based texture".

>So exposing OpenGL related information is necessary.

I never said otherwise. I said that we "should not be binding **the entire API structure** to a specific rendering backend"